### DISCUSSION TITLE
Evaluation on the training set in example notebook

### DISCUSSION THREAD

**Discussion by zachschillaci27** [aurelio-labs/semantic-router#213] [General]
Looking at the example in [06-threshold-optimization.ipynb](https://github.com/aurelio-labs/semantic-router/blob/main/docs/06-threshold-optimization.ipynb), it appears that you are fitting the `RouteLayer` model on a dataset and then evaluating the model on that same dataset. If that's indeed the case, then it's no surprise the accuracy increases dramatically. Did you perhaps also evaluate the trained model on an unseen dataset? I would be curious to know how the performance generalizes. Thanks!

**Comment by ChristianWeyer** [comment]
I actually had the exact same question...
